### PR TITLE
feat/203/Update homepage

### PR DESCRIPTION
**Pull request by CarlosMilano**
# What does this PR do?

Update our current homepage. adds new section called Experience in /profile, this section help collect info to show it in the homepage

# Where should the reviewer start?

app/routes/index.tsx
app/routes/profile/$email/index.tsx

# How should this be manually tested?

Go to /profile
add experience at the bottom
Go to /
Check if your experience is there

# Any background context you want to provide?

# What are the relevant tickets?

https://github.com/wizeline/remix-project-lab/issues/203

# Screenshots

Captura de pantalla 2023-08-24 a la(s) 10 58 50

Captura de pantalla 2023-08-24 a la(s) 10 59 32

Captura de pantalla 2023-08-24 a la(s) 10 59 54

# Questions

# Checklist

- [ ] I added the necessary documentation, if appropriate.
- [ ] I added tests to prove that my fix is effective or my feature works.
- [ ] I reviewed existing Pull Requests before submitting mine.

### CODE REVIEW

File: app/models/experience.server.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    })
+    
+    return result
+}
+
+export async function getExperience() {
+    const result =  prisma.experience.findMany({
+        include: { profile: {select: {avatarUrl: true }}}
+    });
+    return result
```
Comment (martinrobled0) [inline app/models/experience.server.ts:16]: add  ;

File: app/models/experience.server.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+
+import { prisma } from "../db.server";
+
+export async function createExperience(comentario: string, profileId: string){
+    const result = prisma.experience.create({
+        data: { comentario: comentario, profile: { connect: { id: profileId } } },
+    })
+    
+    return result
```
Comment (martinrobled0) [inline app/models/experience.server.ts:9]: add ;

File: app/models/experience.server.ts (hunk 1 above)
Comment (martinrobled0) [reply app/models/experience.server.ts:16]: maybe you missed restart VSC to have the eslint installation complete

File: app/routes/index.tsx [hunk 3]
```diff
@@ -1,120 +1,88 @@
...
-            >
-              Projects
-            </Link>
-          ) : (
-            <Link
-              to="/login"
-              className="flex items-center justify-center rounded-md bg-yellow-500 px-4 py-3 font-medium text-white hover:bg-yellow-600"
+    <article>
+      <HomeHeader>
+        <img src="/wizeletters.png" alt="Wizeline" width={190} />
```
Comment (martinrobled0) [inline app/routes/index.tsx:38]: you can add a more descriptive alt value

File: app/routes/index.tsx [hunk 4]
```diff
@@ -1,120 +1,88 @@
...
-                </Grid>
-              ))}
-            </Grid>
-          </Footer>
-        </StyledLoginForm>
-      </Panel>
-    </LoginPageContainer>
+          <Button
+            href="/projects"
+            className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1a9vgbr-MuiButtonBase-root-MuiButton-root"
```
Comment (martinrobled0) [inline app/routes/index.tsx:75]: why did you add the class name manually?

File: app/routes/profile/$email/index.tsx [hunk 5]
```diff
@@ -9,16 +9,38 @@ import {
...
-import type { LoaderArgs, LoaderFunction } from "@remix-run/node";
+import Card from "@mui/material/Card";
+import CardContent from "@mui/material/CardContent";
+import {
+  getGitHubProfileByEmail,
+  getGitHubProjectsByEmail,
+} from "../../../models/profile.server";
+import { useLoaderData } from "@remix-run/react";
+import { ActionFunction,
+  type LoaderArgs,
```
Comment (martinrobled0) [inline app/routes/profile/$email/index.tsx:23]: here you need to define  LoaderArgs, LoaderFunction as types like this 
`import type { LoaderArgs, LoaderFunction } from "@remix-run/node";`

File: app/routes/profile/$email/index.tsx [hunk 6]
```diff
@@ -102,24 +158,24 @@ export const ProfileInfo = () => {
                     />
                   </Box>
                   <Box textAlign="center">
-                    <h1> { firstName + " " +  lastName }</h1>
-                    <h4 style={{margin:0}}>{ username }</h4>
+                    <h1> {firstName + " " + lastName}</h1>
```
Comment (martinrobled0) [inline app/routes/profile/$email/index.tsx:161]: I think we can replace this with a <Typography> element

**Review by martinrobled0 — CHANGES_REQUESTED** [review]

File: app/models/experience.server.ts (hunk 1 above)
Comment (jackbravo) [reply app/models/experience.server.ts]: yes, in general the changes can be made automatically for you if you install the prettier VSCode extension

**Comment by jackbravo** [comment]
Looking at this... I'm thinking maybe we later we would need an admin page to select which quotes can be shown in the homepage... but I see that just as a nice to have for later, it won't be a problem until we have lots of quotes :-p

**Comment by CarlosMilano** [comment]
It is actually a dynamic home page, it only shows 4 quotes, if you refresh the page you will see 4 different quotes, you can't appreciate this cause there's no info in the db.

**Review by AxelUZ — COMMENTED** [review]
After reviewing the new commits I think it looks great. I tested it by adding a comment, and it showed up on the new home page successfully. I also find it interesting that the page is dynamic showing different comments each time the page is refreshed.

File: app/routes/index.tsx [hunk 7]
```diff
@@ -1,120 +1,88 @@
...
+      <HomeHeader>
+        <img src="/wizeletters.png" alt="Wizeline letters for homepage header" width={190} />
+      </HomeHeader>
+      <HomePageContainer>
+        <img src="/background.jpg" alt="Background fo the homepage" style={{ width: "100%" }} />
+        <Stack
+          direction={{ xs: "column", sm: "row" }}
+          margin={{ xs: "20px", sm: "5px", md: "30px" }}
+        >
+          {!info && "No hay experiencia"}
```
Comment (martinrobled0) [inline app/routes/index.tsx:46]: Maybe we can add another element to cointain the text

File: app/routes/profile/$email/index.tsx [hunk 8]
```diff
@@ -136,6 +195,40 @@ export const ProfileInfo = () => {
...
+                      id="outlined-basic"
+                      label="Your Experience"
+                      variant="outlined"
+                      style={{ width: "100%" }}
+                      name={"comentario"}
+                    ></TextField>
+                    <Grid sx={{ display: "flex", justifyContent: "center" }}>
+                      <Button
+                        type="submit"
+                        className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1a9vgbr-MuiButtonBase-root-MuiButton-root"
```
Comment (martinrobled0) [inline app/routes/profile/$email/index.tsx:218]: the classes still here, it's necessary ?

**Review by martinrobled0 — CHANGES_REQUESTED** [review]

**Review by martinrobled0 — APPROVED** [review]
nice first PR 💯  .. LGTM
